perf/perf_pcp.py

```diff
@@ -0,0 +1,106 @@
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import distro, process, cpu
+from avocado.utils.software_manager import SoftwareManager
+
+
+class PCP(Test):
+
+    """
+    Performance Co Pilot (PCP)
+    1. Starts pmcd daemon
+    2. Install perfevent pmda
+    3. Test pcp command
+    4. Test 24x7 events
+    5. Remove perfevent pmda
+    """
+
+    def setUp(self):
+        smg = SoftwareManager()
+        self.cpu_arch = cpu.get_cpu_arch().lower()
+        dist = distro.detect()
+        if dist.name in ['centos', 'fedora', 'rhel', 'SuSE']:
+            pkgs = ['pcp', 'pcp-pmda-perfevent']
+        else:
+            self.cancel("PCP is not supported on %s" % dist.name)
+
+        for pkg in pkgs:
+            if not smg.check_installed(pkg) and not smg.install(pkg):
+                self.cancel("Package %s is missing/could not be installed"
+                            % pkg)
+
+    def lpar_24x7_ppc64le_check(self):
+        if 'ppc' not in process.system_output("uname -p", ignore_status=True):
+            self.cancel("Processor is not ppc64")
+        # Check if this is a guest
+        # 24x7 is not suported on guest
+        cpu_output = process.run("cat /proc/cpuinfo")
+        if "emulated by" in cpu_output.stdout:
+            self.cancel("24x7: This test is not supported on guest")
+
+        # Check if 24x7 is present
+        if not os.path.exists("/sys/bus/event_source/devices/hv_24x7"):
+            self.cancel("%s doesn't exist.This feature is supported"
+                        " only on LPAR" % self.event_sysfs)
+
+    def test_pmcd_daemon(self):
+        output = process.run("systemctl start pmcd", shell=True)
+        output = process.run("systemctl status pmcd", shell=True)
+        if "active (running)" not in output.stdout:
+            self.fail("PCP: Can not start pmcd daemon")
+
+    def test_pmda_perfevent_install(self):
+        if not os.path.isfile("/var/lib/pcp/pmdas/perfevent/Install"):
+            self.fail("PCP: perfevent pmda package not installed")
+        cmd = "cd /var/lib/pcp/pmdas/perfevent/; echo pipe | ./Install"
+        output = process.run(cmd, shell=True)
+        output = process.run("systemctl restart pmcd", shell=True)
+        output = process.run("systemctl status pmcd", shell=True)
+        if "active (running)" and "pmdaperfevent" not in output.stdout:
+            self.fail("PCP: perfevent pmda not reflected in pmcd daemon")
+
+    def test_pcp_cmd(self):
+        output = process.run("pcp", shell=True)
+        if "Cannot connect to PMCD" in output.stdout:
+            self.fail("PCP: pmcd daemon dead")
+
+    def test_perfevent_24x7_events(self):
+        self.lpar_24x7_ppc64le_check()
+        if self.cpu_arch == 'power8':
+            self.cancel("Not supported on Power8")
+        # -s 2 collects two samples
+        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC.value"
+        output = process.run(cmd, shell=True)
```

Comment:
`output` variable not required?

Reply:
@harish-24 `output` content is not required.